### PR TITLE
Account abstraction

### PR DESCRIPTION
**Pull request by villanuevawill**
Summary: Includes tests, logic and behavior around an account model being committed to the block with associated error checking/value updates.

Please note that I changed the old Account struct name to EthAccount as it is mainly a struct that exists for testing.

### CODE REVIEW

**Comment by villanuevawill** [comment]
TODO on a future PR. Tests can start being cleaned up. There's a lot of repetitive behavior right now in the test suites. :)

**Comment by villanuevawill** [comment]
NEXT thing to tackle: gas logic and mempool.

File: core/src/account.rs [hunk 1]
```diff
@@ -9,29 +10,47 @@ pub type PrivateKey = H256;
...
+    pub balance: U256,
+}
+
+impl Account {
+    pub fn new(balance: U256) -> Account {
+        Account { balance }
+    }
+}
+
+impl Clone for Account {
```
Comment (c-o-l-o-r) [inline core/src/account.rs]: Did you try using `#[derive(clone)]`? I would expect that to perform the same function, but more succinctly.

File: core/src/transaction.rs [hunk 2]
```diff
@@ -43,41 +44,53 @@ pub struct SignedTransaction {
     pub signature: Signature,
 }
 
+impl Deref for SignedTransaction {
```
Comment (c-o-l-o-r) [inline core/src/transaction.rs]: nice

File: core/src/account.rs (hunk 1 above)
Comment (villanuevawill) [reply core/src/account.rs]: Awesome - and done! :)

**Pull request merged**